src/authorization/AclOwnershipReader.ts

```diff
@@ -0,0 +1,52 @@
+import { CredentialGroup } from '../authentication/Credentials';
+import type { AccountSettings, AccountStore } from '../identity/interaction/email-password/storage/AccountStore';
+import type { AuxiliaryIdentifierStrategy } from '../ldp/auxiliary/AuxiliaryIdentifierStrategy';
+import type { AclPermission } from '../ldp/permissions/AclPermission';
+import type { PermissionSet } from '../ldp/permissions/Permissions';
+import type { PermissionReaderInput } from './PermissionReader';
+import { PermissionReader } from './PermissionReader';
+
+/**
+ * Allows full access on ACL resources if the request is being made by the owner of the pod containing that resource.
+ */
+export class AclOwnershipReader extends PermissionReader {
+  private readonly accountStore: AccountStore;
+  private readonly aclStrategy: AuxiliaryIdentifierStrategy;
+
+  public constructor(accountStore: AccountStore, aclStrategy: AuxiliaryIdentifierStrategy) {
```

Comment:
Maybe logging would be interesting; could get confusing otherwise.

Reply:
Implementation suggestion (if you think it makes sense): there are currently multiple paths returning `{}`. Perhaps these contents could instead be in a helper function that throws the exact reason for not granting the permissions; then the main `handle` catches the error, logs it as a `debug`, and returns `{}`. Then you could eliminate the comments from the function as well, as they would essentially be the error messages. (But that also holds if you just log them.)

Reply:
It's something I also wasn't quite sure of, both ways have some merit. The reason why it was done like this is that all other PermissionReaders can handle all input. Also because this one can handle the input, it just provides no permissions. But it's definitely not clearcut here imo.

Reply:
> The reason why it was done like this is that all other PermissionReaders can handle all input.

No no, definitely keep that!

What I meant is:

```javascript
function canHandle() {
  try {
    this.verifyPodOwner();
  }
  catch (error) {
    this.logger.debug(`No pod owner Control permissions: ${error.message}`);
    return {};
  }
  return controlPermissions;
}
```

where `verifyPodOwner` contains the logic currently in `handle`, but then throws an explicit reason instead of returning `{}`:

```typescript
  public async handle({ credentials, identifier }: PermissionReaderInput): Promise<PermissionSet> {
    if (!this.aclStrategy.isAuxiliaryIdentifier(identifier)) {
      throw new Error("Not an ACL resource");
    }
```

So `handle` itself never throws.

Reply:
Ah sure, that works (I'm assuming you meant to write `function handle(input)` above).

Reply:
Just a suggestion though. (Indeed!)

Reply:
Just a suggestion though. (Indeed!)